### PR TITLE
feat(dh): library generator part 2

### LINKED ISSUE #722: Generator for different types of libraries (part 2)
Follow-up to #652

**Description**
Create a generator for different libraries to make it easy to set up a specific library
So that we simplify how to configure each type individually.

The libraries to support is listed here: https://github.com/Energinet-DataHub/greenforce-frontend/blob/main/README.md#library-types

**AC**
- Should expand the existing generator to support the following library types:
  - [ ] ui
  - [ ] routing
  - [ ] domain
  - [ ] configuration
  - [ ] environments
  - [ ] util
  - [ ] test-util
  - [ ] e2e-util
- [ ] Document how to use the generator
- [ ] Should update `test-setup.ts` file to correctly setup TestBed specific to the library type
- [ ] Should update README.md file to remove auto-generated text

**Tech notes**
- See Nx repo for inspiration on how to test the generator. [Example](https://github.com/nrwl/nx/blob/master/packages/angular/src/generators/library/library.spec.ts)

**Note**
- `assets` and `styles` types are not in scope

### PR DESCRIPTION
**Pull request by ManBearTM**
## Description

This adds generators for the remaining library types. 

## References

Closes #722

### CODE REVIEW

File: tools/generators/dh-library-generator/index.ts [hunk 1]
```diff
@@ -86,6 +108,22 @@ export default async function (tree: Tree, schema: DhLibrarySchema) {
...
+  } else if (libType === LibraryType.routing) {
+    // Necessary step since the libraryGenerator does not support
+    // generating modules with routing suffix without the --skipImport
+    // flag, which does not currently work with workspace-generators:
+    // https://github.com/nrwl/nx/pull/10167#issuecomment-1126146451
+    await angularMoveGenerator(tree, {
+      updateImportPath: true,
+      projectName: `dh-${libDomain}-routing-tmpl`,
+      destination: `dh/${libDomain}/routing`,
+    });
```
Comment (dzhavat) [inline tools/generators/dh-library-generator/index.ts:120]: praise: nice workaround :)

File: tools/generators/dh-library-generator/index.ts [hunk 2]
```diff
@@ -238,12 +287,70 @@ function exposeEmptyStoreFromLibrary(
...
+  tree: Tree,
+  options: {
+    libPath: string;
+    libDomain: string;
+    className: string;
+  }
+) {
+  const indexPath = `${options.libPath}/src/index.ts`;
+  const storeFileName = `dh-${options.libDomain}-path`;
+  const content = `export * from './lib/${storeFileName}';\n`;
```
[the remark this answers is not in the document]
Comment (ManBearTM) [reply tools/generators/dh-library-generator/index.ts]: Nice one!

File: tools/generators/dh-library-generator/schema.json [hunk 3]
```diff
@@ -26,7 +38,8 @@
     },
     "name": {
       "type": "string",
-      "description": "Library name (note 1: if the selected library type is \"feature\" or \"data-access\", the final name will be prefixed with it, e.g. `feature-<name>`) (note 2: leave this field empty if the selected library type is \"shell\")"
+      "description": "Library name (note 1: if the selected library type is \"configuration\", \"data-access\", \"e2e-util\", \"feature\", \"test-util\", \"ui\" or \"util\", the final name will be prefixed with it, e.g. `feature-<name>`)",
```
[the remark this answers is not in the document]
Comment (ManBearTM) [reply tools/generators/dh-library-generator/schema.json:41]: Good catch, forgot about the `validateParams` function. Also didn't think about the VS Code extension, I will add the note back 🙂

File: tools/generators/dh-library-generator/schema.json [hunk 4]
```diff
@@ -26,7 +38,8 @@
...
+      "x-prompt": "Enter library name (leave blank for \"domain\", \"environments\", \"routing\" and \"shell\")"
```
[the remark this answers is not in the document]
Comment (ManBearTM) [reply tools/generators/dh-library-generator/schema.json:42]: 5edca79285bd1e0237c8072743860a025eb59cf0

File: README.md [hunk 5]
```diff
@@ -159,7 +159,7 @@ Allowed dependencies to: `assets`, `styles`.
 
 To generate a new library, run `yarn nx workspace-generator dh-library-generator` and follow the instructions or use the "workspace-generator - dh-library-generator" option under "generate" command in Nx Console extension.
 
-_Note_: This command currently only supports generating libraries of type _"feature"_, _"data-access"_ and _"shell"_.
+_Note_: This command currently does **not** support generating libraries of type _"assets"_ and _"styles"_.
```
[the remark this answers is not in the document]
Comment (ManBearTM) [reply README.md]: f8bdd243971b8af53ca58f919fcd0c6339dee943

File: tools/generators/dh-library-generator/schema.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (ManBearTM) [reply tools/generators/dh-library-generator/schema.json:41]: 5edca79285bd1e0237c8072743860a025eb59cf0

[the remark this answers is not in the document]
Comment (ManBearTM) [reply tools/generators/dh-library-generator/schema.json]: 76f66b7d16f8db90422bbcaf17e73e3db0b69bca

**Review by dzhavat — APPROVED** [review]

**Pull request merged**